src/routes/components/collapsible/+page.md

```diff
@@ -0,0 +1,125 @@
+</script>
+
+# Collapsible
+
+Collapsible is an accordion element that expands when clicked on. They allow you to hide content that is not immediately relevant to the user. Buttons or anchors are used as triggers that are mapped to specific elements you toggle.
+
+# Usage
+
+```html
+<Collapsible iconSize="medium">
+  <div slot="header">About smart contracts</div>
+  <p>
+    Advanced smart contracts process HTTP requests, control other chains, and
+    scale infinitely.
+  </p>
+</Collapsible>
+```
+
+#### With an external button
+
+```html
+<script>
+  let toggleButton: HTMLElement;
+  let toggleFunction: function;
+</script>
+
+<button class="secondary toggleButton"
+  bind:this={toggleButton}
+  on:click={() => toggleContentF()}
+>
+  Press to open
+</button>
+
+<Collapsible iconSize="medium" expandButton={false} bind:toggleContent={toggleFunction}>
+  <div slot="header" class="header_div" id="header2">
+    Collapsible with external button
+  </div>
+  <div class="content_text">
+    This collapsible uses an external button element to toggle open its content.
+  </div>
+</Collapsible>
+```
+
+## Properties
+
+| Property           | Description                                                     | Type                    | Default     |
+| ------------------ | --------------------------------------------------------------- | ----------------------- | ----------- |
+| `id`               | Corresponding id of the content element, for selection purposes | `string` or `undefined` | `undefined` |
+| `initallyExpanded` | Initially collapsed or expanded                                 | `boolean`               | `false`     |
+| `maxContentHeight` | Optional height that defines the maximum content height         | `number` or `undefined` | `undefined` |
+| `iconSize`         | Size of the icon: either `small` or `medium`                    | `string`                | `small`     |
+| `expandButton`     | Whether to include an expand button                             | `boolean`               | `true`      |
+| `externalToggle`   | To have an external toggle element                              | `boolean`               | `false`     |
+| `wrapHeight`       | To have wrapped height                                          | `boolean`               | `false`     |
+
+## Slots
+
+| Slot name | Description                                                                              |
+| --------- | ---------------------------------------------------------------------------------------- |
+| `header`  | Summary header of the collapsible, usually displayed with an anchor button on the right. |
+
+## Events
+
+| Event           | Description                                                           |
+| --------------- | --------------------------------------------------------------------- |
+| `toggleContent` | A toggle content function that could be bound to an external element. |
+
+## Showcase
+
+<div id="container">
```

Comment:
We don't use`id` for styling purpose because they need to be unique.

Reply:
Can I change this to class name then?

Reply:
Sure class name is fine.